3章で使用したデータ（student-mat.csv及びsutudent-por.csv）で数字とポルトガル語の成績のＧ２をそれ其れの平均について、差があるといえるでしょうか。Ｇ３はどうでしょうか。

In [1]:
import pandas as pd
from scipy import stats

# データ読み込み
student_data_math = pd.read_csv("student/student-mat.csv", sep=";")
student_data_por = pd.read_csv("student/student-por.csv", sep=";")

# 同じ生徒を対応させるための結合キー
merge_keys = [
    "school", "sex", "age", "address", "famsize", "Pstatus",
    "Medu", "Fedu", "Mjob", "Fjob", "reason", "nursery", "internet"
]

# 数学データとポルトガル語データを結合
student_data_merge = pd.merge(
    student_data_math,
    student_data_por,
    on=merge_keys,
    suffixes=("_math", "_por")
)

print("結合後の人数:", len(student_data_merge))

# G2の対応のあるt検定
t_g2, p_g2 = stats.ttest_rel(
    student_data_merge["G2_math"],
    student_data_merge["G2_por"]
)

print("G2 数学平均:", student_data_merge["G2_math"].mean())
print("G2 ポルトガル語平均:", student_data_merge["G2_por"].mean())
print("G2 t値:", t_g2)
print("G2 p値:", p_g2)

# G3の対応のあるt検定
t_g3, p_g3 = stats.ttest_rel(
    student_data_merge["G3_math"],
    student_data_merge["G3_por"]
)

print("G3 数学平均:", student_data_merge["G3_math"].mean())
print("G3 ポルトガル語平均:", student_data_merge["G3_por"].mean())
print("G3 t値:", t_g3)
print("G3 p値:", p_g3)

結合後の人数: 382
G2 数学平均: 10.712041884816754
G2 ポルトガル語平均: 12.238219895287958
G2 t値: -9.431186078657426
G2 p値: 4.0622824801348495e-19
G3 数学平均: 10.387434554973822
G3 ポルトガル語平均: 12.515706806282722
G3 t値: -9.977077139661143
G3 p値: 5.5614921136883726e-21


有意水準5%で対応のあるt検定を行った。

G2について、数学の平均は10.712、ポルトガル語の平均は12.238であった。
検定の結果、p値は4.06×10^-19となり、0.05より小さかった。
したがって、G2の平均には差があると言える。

G3について、数学の平均は10.387、ポルトガル語の平均は12.516であった。
検定の結果、p値は5.56×10^-21となり、0.05より小さかった。
したがって、G3の平均にも差があると言える。